unittests/rpc/handlers/NFTInfoTest.cpp

```diff
@@ -0,0 +1,449 @@
+// idk why this case will happen in reality
+TEST_F(RPCNFTInfoHandlerTest, NonExistLedgerViaLedgerHash2)
+{
+    MockBackend* rawBackendPtr =
+        static_cast<MockBackend*>(mockBackendPtr.get());
+    mockBackendPtr->updateRange(10);  // min
+    mockBackendPtr->updateRange(30);  // max
+    // mock fetchLedgerByHash return ledger but seq is 31 > 30
+    auto ledgerinfo = CreateLedgerInfo(LEDGERHASH, 31);
+    ON_CALL(*rawBackendPtr, fetchLedgerByHash(ripple::uint256{LEDGERHASH}, _))
+        .WillByDefault(Return(ledgerinfo));
+    EXPECT_CALL(*rawBackendPtr, fetchLedgerByHash).Times(1);
+    auto const input = json::parse(fmt::format(
+        R"({{ 
+            "nft_id": "{}",
+            "ledger_hash": "{}"
+        }})",
+        NFTID,
+        LEDGERHASH));
+    runSpawn([&, this](boost::asio::yield_context yield) {
+        auto const handler = AnyHandler{NFTInfoHandler{mockBackendPtr}};
+        auto const output = handler.process(input, yield);
+        ASSERT_FALSE(output);
+        auto const err = RPC::makeError(output.error());
+        EXPECT_EQ(err.at("error").as_string(), "lgrNotFound");
+        EXPECT_EQ(err.at("error_message").as_string(), "ledgerNotFound");
+    });
+}
+
+// error case ledger > max seq via index
+TEST_F(RPCNFTInfoHandlerTest, NonExistLedgerViaLedgerIndex2)
+{
+    MockBackend* rawBackendPtr =
+        static_cast<MockBackend*>(mockBackendPtr.get());
+    mockBackendPtr->updateRange(10);  // min
+    mockBackendPtr->updateRange(30);  // max
+    // no need to check from db,call fetchLedgerBySequence 0 time
+    // differ from previous logic
+    EXPECT_CALL(*rawBackendPtr, fetchLedgerBySequence).Times(0);
+    auto const input = json::parse(fmt::format(
+        R"({{ 
+            "nft_id": "{}",
+            "ledger_index": "31"
+        }})",
+        NFTID));
+    runSpawn([&, this](boost::asio::yield_context yield) {
+        auto const handler = AnyHandler{NFTInfoHandler{mockBackendPtr}};
+        auto const output = handler.process(input, yield);
+        ASSERT_FALSE(output);
+        auto const err = RPC::makeError(output.error());
+        EXPECT_EQ(err.at("error").as_string(), "lgrNotFound");
+        EXPECT_EQ(err.at("error_message").as_string(), "ledgerNotFound");
+    });
+}
+
+// error case nft does not exist
+TEST_F(RPCNFTInfoHandlerTest, NonExistNFT)
+{
+    MockBackend* rawBackendPtr =
+        static_cast<MockBackend*>(mockBackendPtr.get());
+    mockBackendPtr->updateRange(10);  // min
+    mockBackendPtr->updateRange(30);  // max
+    auto ledgerinfo = CreateLedgerInfo(LEDGERHASH, 30);
+    ON_CALL(*rawBackendPtr, fetchLedgerByHash(ripple::uint256{LEDGERHASH}, _))
+        .WillByDefault(Return(ledgerinfo));
+    EXPECT_CALL(*rawBackendPtr, fetchLedgerByHash).Times(1);
+    // fetch nft return emtpy
+    ON_CALL(*rawBackendPtr, fetchNFT)
+        .WillByDefault(Return(std::optional<NFT>{}));
+    EXPECT_CALL(*rawBackendPtr, fetchNFT).Times(1);
```

Comment:
I suggest we can add matcher here to verify the handler calls fetchNFT correctly. Without it , we can not catch the error like token is parsed incorrectly or not use the specific ledger seq